.github/workflows/release.yml

```diff
@@ -3,60 +3,109 @@ name: Release
 on:
   workflow_dispatch:
     inputs:
-      deploy-to:
-        type: choice
-        description: Choose where to publish (test/prod)
-        options:
-        - prod
-        - test
-        default: prod
-      ref:
-        description: "The ref (sha or branch name) to use"
+      branch:
+        description: "The branch to release from"
         type: string
         default: "main"
+      version:
+        description: "The version to release"
         required: true
+        type: string
+      deploy-to:
+        description: "Deploy to test or prod"
+        type: environment
+        default: prod
+      only_docker:
+        description: "Only release Docker image, skip GitHub & PyPI"
+        type: boolean
+        default: false
 
-permissions: read-all
+permissions:
+  contents: write # this is the permission that allows creating a new release
 
 # will cancel previous workflows triggered by the same event and for the same ref for PRs or same SHA otherwise
 concurrency:
-  group: ${{ github.workflow }}-${{ github.event_name }}-${{ contains(github.event_name, 'pull_request') && github.event.pull_request.head.ref || github.sha }}-${{ inputs.deploy-to }}
+  group: "${{ github.workflow }}-${{ github.event_name }}-${{ inputs.version }}-${{ inputs.deploy-to }}"
   cancel-in-progress: true
 
 jobs:
-  release:
-    name: PyPI - ${{ inputs.deploy-to }}
-    runs-on: ubuntu-latest
-    environment:
-      name: ${{ inputs.deploy-to }}
-      url: ${{ vars.PYPI_PROJECT_URL }}
-    permissions:
-      id-token: write  # IMPORTANT: this permission is mandatory for trusted publishing
+  release-prep:
+    name: "Release prep: generate changelog, bump version"
+    uses: dbt-labs/dbt-postgres/.github/workflows/release_prep_hatch.yml@main
+    with:
+      branch: ${{ inputs.branch }}
+      version: ${{ inputs.version }}
+      deploy-to: ${{ inputs.deploy-to }}
+    secrets: inherit
 
+  build-release:
+    name: "Build release"
+    needs: release-prep
+    runs-on: ubuntu-latest
+    outputs:
+      archive-name: ${{ steps.archive.outputs.name }}
     steps:
-      - name: Check out repository
+      - name: "Checkout ${{ github.event.repository.name }}@${{ needs.release-prep.outputs.release-branch }}"
```

Comment:
Oh geez is the output param implied from the reusable job call? That's .... funky

Reply:
Yeah, that's how all reusable workflows/actions work. I agree it's a bit cumbersome, but I don't have a better suggestion.